### PR TITLE
dashboard: restructure packages in pkg/

### LINKED ISSUE #2238: dashboard: restructure packages in pkg/
As https://github.com/chaos-mesh/chaos-mesh/pull/2218#issuecomment-900781325 described, we will open a new PR to solve this problem.

### PR DESCRIPTION
**Pull request by fewdan**
Signed-off-by: “fewdan” <fewdan@hotmail.com>

### What problem does this PR solve?

Issue Number: close #2238  

Problem Summary:

### What is changed and how it works?

I tested it manually. Can you see what else needs to be modified?  @STRRL  @g1eny0ung 

What's Changed:

### Related changes

* PR to update `chaos-mesh/website`/`chaos-mesh/website-zh`:
* Need to update Chaos Dashboard component, related issue:
* Need to cheery-pick to the release branch

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Breaking backward compatibility

### Release note 

```release-note
Please add a release note.
If you don't think this PR needs a release note then fill it with None.
```

### CODE REVIEW

**Comment by STRRL** [comment]
hi @fewdan, please resolve the conflicts.

**Comment by fewdan** [comment]
> hi @fewdan, please resolve the conflicts.

@STRRL done.  PTAL, Thanks.

File: .gitignore [hunk 1]
```diff
@@ -32,7 +32,10 @@ _artifacts/
 e2e-test/image/e2e/chaos-mesh
 e2e-test/image/e2e/manifests/
 
-/pkg/uiserver/embedded_assets_handler.go
+/pkg/dashboard/uiserver/embedded_assets_handler.go
+
+pkg/dashboard/swaggerdocs/*
+!pkg/dashboard/swaggerdocs/placeholder.go
```
Comment (STRRL) [inline .gitignore:38]: @YangKeao @g1eny0ung Should we do that?

**Review by STRRL — COMMENTED** [review]
That's awesome!

rest LGTM!

File: .gitignore (hunk 1 above)
Comment (YangKeao) [reply .gitignore:38]: no go code should be ignored. Ignoring go codes will drop dependencies in `go.sum` and `go.mod`.

Comment (fewdan) [reply .gitignore:38]: Discussed with @YangKeao , we need to keep all the go code, so here I am going to modify it.

**Review by STRRL — APPROVED** [review]
LGTM!

**Review by g1eny0ung — CHANGES_REQUESTED** [review]
image

As described in the comments above, we should filter out files other than `placeholder.go` (which is automatically generated).

**Comment by STRRL** [comment]
> image
> 
> As described in the comments above, we should filter out files other than `placeholder.go` (which is automatically generated).

If I understand correctly, `docs.go` also should be included, and the import of `github.com/alecthomas/template` has been always an issue for a long time after building with `SWAGGER=1`

**Comment by STRRL** [comment]
As all of the `docs.go`, `swagger.json` and `swagger.yaml` have similar content, it's OK for me to commit them into the git repo, just like we commit the crd.yaml into git repo.

**Comment by g1eny0ung** [comment]
> If I understand correctly, `docs.go` also should be included, and the import of `github.com/alecthomas/template` has been always an issue for a long time after building with `SWAGGER=1`

@STRRL The missing `github.com/alecthomas/template` is indeed a long-standing issue. Especially in CI.

But I think it's also a regular practice to ignore the generated files. And even after changing the swagger definition in the API server, this part needs to be generated by running the command manually. We should be able to make sure it's up to date if we want to keep track of it in the repository.

Otherwise, it would be irresponsible for us to throw these things in the repo.

**Comment by STRRL** [comment]
> > If I understand correctly, `docs.go` also should be included, and the import of `github.com/alecthomas/template` has been always an issue for a long time after building with `SWAGGER=1`
> 
> @STRRL The missing `github.com/alecthomas/template` is indeed a long-standing issue. Especially in CI.
> 
> But I think it's also a regular practice to ignore the generated files. And even after changing the swagger definition in the API server, this part needs to be generated by running the command manually. We should be able to make sure it's up to date if we want to keep track of it in the repository.
> 
> Otherwise, it would be irresponsible for us to throw these things in the repo.

Oh, how about appending recipe `swagger_spec` as a dependency for recipe `check` in `Makefile`? @g1eny0ung @fewdan @YangKeao

**Comment by g1eny0ung** [comment]
> Oh, how about appending recipe swagger_spec as a dependency for recipe check in Makefile?

That would be great.

**Comment by fewdan** [comment]
> > > If I understand correctly, `docs.go` also should be included, and the import of `github.com/alecthomas/template` has been always an issue for a long time after building with `SWAGGER=1`
> > 
> > 
> > @STRRL The missing `github.com/alecthomas/template` is indeed a long-standing issue. Especially in CI.
> > But I think it's also a regular practice to ignore the generated files. And even after changing the swagger definition in the API server, this part needs to be generated by running the command manually. We should be able to make sure it's up to date if we want to keep track of it in the repository.
> > Otherwise, it would be irresponsible for us to throw these things in the repo.
> 
> Oh, how about appending recipe `swagger_spec` as a dependency for recipe `check` in `Makefile`? @g1eny0ung @fewdan @YangKeao

So, in the check part of the makefile, execute "swagger_spec"?

**Comment by YangKeao** [comment]
> > > If I understand correctly, `docs.go` also should be included, and the import of `github.com/alecthomas/template` has been always an issue for a long time after building with `SWAGGER=1`
> > 
> > 
> > @STRRL The missing `github.com/alecthomas/template` is indeed a long-standing issue. Especially in CI.
> > But I think it's also a regular practice to ignore the generated files. And even after changing the swagger definition in the API server, this part needs to be generated by running the command manually. We should be able to make sure it's up to date if we want to keep track of it in the repository.
> > Otherwise, it would be irresponsible for us to throw these things in the repo.
> 
> Oh, how about appending recipe `swagger_spec` as a dependency for recipe `check` in `Makefile`? @g1eny0ung @fewdan @YangKeao

It's already a dependency of `check` (because it's the dependency of `generate`, which is the dependency of `check`). The problem is that it's not marked as `PHONY`. I will submit a PR to fix this.

**Review by g1eny0ung — APPROVED** [review]

**Review by STRRL — APPROVED** [review]
LGTM!

**Pull request merged**